src/components/PageTitle.js

```diff
@@ -0,0 +1,12 @@
+import React, { useEffect } from 'react';
+import {observer} from "mobx-react";
+import {autorun} from 'mobx';
+import {useAnswersState} from "../state/AnswersStateProvider";
+
+export const PageTitle = observer(() => {
+    const answers = useAnswersState();
+    useEffect(() => autorun(() => {
+        document.title = answers.isLoading ? 'Loading...' : `Competitions: ${answers.allCount}`;
+    }), []);
```

Comment:
This is what I don't like actually. 

`useEffect(() => {
  ...
}, [answers.isLoading])`

Reply:
I agree... I also don't like. What I was looking for is the way to use mobx side effects in React somehow..

Reply:
Yeap, I know. 

We could defend somehow that in Mobx, Mobx is taking care of reactivity. There are some benefits of that, e.g. performance. Also, we could say that we're presenting Mobx as library and it can be use without React. I'd perfect to Selfridges case when we really struggling with reactivity. Do you remember? :) I wasn't so easily with pure Redux and Reselect. 

But in React, this syntax sucks. And there is also another downside of this. The closure for autorun is created on component mount. So if you use any other variables like state or props, you are going to reference the old values. I'm going to deep dive into this one.

Reply:
So this one mate: https://github.com/mobxjs/mobx-react/issues/772

This conversation is really mind blowing but they come up with pretty the same thoughts: useEffect.

Reply:
772 is getting famous and it is references in other conversations: https://github.com/mobxjs/mobx-react/issues/853